website/public/js/controllers/groupsCtrl.js

```diff
@@ -224,6 +224,9 @@ habitrpg.controller("GroupsCtrl", ['$scope', '$rootScope', 'Shared', 'Groups', '
       if ($scope.query === undefined || $scope.query === null) {
         return false;
       }
+      if (msg.username == undefined) {
+        return false;
```

Comment:
Actually, the test is based on $scope.query.test. So maybe:

if (!$scope.query || !msg.username || $scope.query.text !== undefined) {
      return false;
}

Is there a good way to test if something is a string in JavaScript? Because that is actually what we want to test for msg.username and $scope.query.text